edb/edgeql/compiler/options.py

```diff
@@ -116,13 +117,15 @@ class CompilerOptions(GlobalCompilerOptions):
 
     #: A set of schema types and links that should be treated
     #: as singletons in the context of this compilation.
-    singletons: Collection[Union[s_types.Type, s_pointers.Pointer]] = (
-        frozenset())
+    singletons: Collection[
+        s_types.Type | s_pointers.Pointer | s_constraints.ConsistencySubject
```

Comment:
I think that all ConsistencySubjects are already `Type` or `Pointer`. I'd add an assert at the callsite instead of changing the type here (and then would try to not make the downstream changes to type_remaps and view_sets?)